src/rules/extensions.js

```diff
@@ -190,3 +191,39 @@ module.exports = {
     return moduleVisitor(checkFileExtension, { commonjs: true });
   },
 };
+
+/**
+ * Taken from `eslint-import-resolver-typescript`.
+ * This could be imported from current versions of that plugin,
+ * but this project still depends on an older version.
+ * Also, importing it would add a dependency, or at least an
+ * optional peer dependency - copying the code seems like the
+ * more sane option.
+ * [LICENSE](https://github.com/import-js/eslint-import-resolver-typescript/blob/71b23a206514842fef70a99220e5ffb1d6da2a0e/LICENSE)
+ */
```

Comment:
is it that we depend on an older *major* of that plugin? or that we depend on an older minor or patch?

Probably a better option would be extracting the code into its own package, that both we and eslint-import-resolver-typescript use. @JounQin, thoughts?

Reply:
`eslint-plugin-import` doesn't have any `dependency` on `eslint-import-resolver-typescript` yet, but a `devDependency` on `1.0.0 || 1.1.0` - the current version is `3.5.5`.  
I didn't want to add a new dependency here just for that one config object, but I'm happy with whatever solution we end up with here :)

Reply:
ah, true enough. adding a dependency is fine, but perhaps the resolver's not ideal to add. hopefully an extracted package is viable.

Reply:
If that doesn't work, we could go for an `peerDependency` with `peerDependenciesMeta` `optional: true`. 
If the package is there, use it, if it isn't we should avoid the relevant codeblock anyways.

Reply:
I don't want to rely on optional peers; any npm client older than that feature would treat it as required.

Reply:
Good point. Then let's see what @JounQin has to say :)

Reply:
Should it be a setting of import plugin to for using without TypeScript resolver? For example, could webpack resolver resolves same path?

Reply:
I think that's already the `settings['import/resolver'].typescript` setting? 
If that's set, the import plugin will use the TS resolver, otherwise not.

Reply:
https://webpack.js.org/configuration/resolve/#resolveextensionalias

I mean the webpack resolver could also resolve different extensions if it's working correctly? Then only relying on the typescript resolver's setting seems incorrect to me.

Reply:
Maybe? I don't know what that one does, or if anyone ever had this problem with it.  
The problem at hand is that this is something TypeScript expects with the `node16` target, for at least the last 10 releases, and it is not supported by the `imports/extensions` rule. There doesn't seem any way of asking the resolver "is this a valid replacement file extension for this other extension", since the resolver is pretty much hidden from the rule, so I don't know a way of implementing this better.